### PR TITLE
fix quantize tools cross platform issues

### PR DESCRIPTION
**Pull request by kalcohol**
1. for cross platform, use cv::CommandLineParser instead of getopt.h
2. use cv::glob to search folder instead of dirent.h
3. fix some other warnings of function from cstdlib
4. add some const and some auto

now the tools is usable.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/ncnn?pullRequest=1505) <br/>All committers have signed the CLA.

**Comment by kalcohol** [comment]
oh no, all CI do not has opencv......

File: tools/quantize/ncnn2int8.cpp [hunk 1]
```diff
@@ -505,16 +541,16 @@ int NetQuantize::save(const char* parampath, const char* binpath)
         ncnn::ParamDict pd;
         layer_default->load_param(pd);
 
-#define fprintf_param_value(format, phase) \
+#define FPRINTF_PARAM_VALUE(format, phase) \
```
Comment (nihui) [inline tools/quantize/ncnn2int8.cpp]: keep the lowercase style macro

Comment (kalcohol) [reply tools/quantize/ncnn2int8.cpp]: solved.

**Comment by kalcohol** [comment]
why still show 1 change requested...

**Comment by nihui** [comment]
keep the one line style block

```
            { if (!op->starts.empty()) fprintf_param_int_array(9, op->starts, pp); }
```
```
            {
                if (!op->starts.empty()) fprintf_param_int_array(9, op->starts, pp);
            }
```

**Comment by pineking** [comment]
we use quantize tools without opencv, that's not a good idea for using these opencv dependencies.

**Comment by kalcohol** [comment]
> we use quantize tools without opencv, that's not a good idea for using these opencv dependencies.

um, which image decoder did u chosen?

**Comment by pineking** [comment]
> > we use quantize tools without opencv, that's not a good idea for using these opencv dependencies.
> 
> um, which image decoder did u chosen?

ignore it, that's my mistake, for offline tools, such as quantize, benchmark. actually, we also use opencv to read image files.

**Pull request merged**